src/component/DashboardTable/OrganizationGrid.tsx

```diff
@@ -0,0 +1,374 @@
+	justifyContent: "center",
+	padding: "20px 0px 10px 0px",
+});
+
+const Search = styled("div")(({ theme }) => ({
+	backgroundColor: "#fff",
+	padding: "5px 15px",
+	borderRadius: theme.shape.borderRadius,
+	minWidth: "30%",
+	width: "auto",
+	border: "1px solid black",
+	display: "flex",
+	justifyContent: "space-between",
+	alignItems: "center",
+	gap: "10px",
+}));
+
+const StyledModal = styled(Modal)({
+	display: "flex",
+	justifyContent: "center",
+	alignItems: "center",
+});
+
+function createData(
+	id: number,
+	name: string,
+	tagline: string,
+	banner: string,
+	organizationURL: string,
+	borderRadius: string,
+	buttonColors: string,
+	authMethod: string
+): RowData {
+	return {
+		id,
+		name,
+		tagline,
+		banner,
+		organizationURL,
+		borderRadius,
+		buttonColors,
+		authMethod,
+	};
+}
+
+export default function TableGrid() {
+	const [rows, setRows] = useState<RowData[]>([]);
+	const [searchQuery, setSearchQuery] = useState<string>("");
+	const [page, setPage] = useState<number>(1);
+	const [open, setOpen] = useState(false);
+	const [selectedRow, setSelectedRow] = useState<RowData | null>(null);
+	const [editRowIndex, setEditRowIndex] = useState<number | null>(null);
+
+	useEffect(() => {
+		const fetchOrganizationData = async () => {
+			try {
+				const response = await axios.get("http://localhost:3000/organizations");
+				const organizationData: Organization[] = response.data;
+				const initialRows: RowData[] = organizationData.map((organization) =>
+					createData(
+						organization.id,
+						organization.name,
+						organization.tagline,
+						organization.banner,
+						organization.organizationURL,
+						organization.borderRadius,
+						organization.buttonColors,
+						organization.authMethod
+					)
+				);
+				setRows(initialRows);
+			} catch (error) {
+				console.error("Error fetching organization data:", error);
+			}
+		};
+
+		fetchOrganizationData();
+	}, []);
+
+	const itemsPerPage = 5;
+
+	const filteredRows = rows.filter((row) =>
+		row.name.toLowerCase().includes(searchQuery.toLowerCase())
+	);
+
+	const startIndex = (page - 1) * itemsPerPage;
+	const endIndex = startIndex + itemsPerPage;
+	const paginatedRows = filteredRows.slice(startIndex, endIndex);
+
+	const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+		setSearchQuery(event.target.value);
+		setPage(1);
+	};
+
+	const handlePageChange = (
+		event: React.ChangeEvent<unknown>,
+		value: number
+	) => {
+		setPage(value);
+	};
+
+	const handleRemoveOrg = async (id: number) => {
+		try {
+			await axios.delete(`http://localhost:3000/organizations/${id}`);
+			const updatedRows = rows.filter((row) => row.id !== id);
+			setRows(updatedRows);
+		} catch (error) {
+			console.error("Error deleting organization:", error);
+		}
+	};
+
+	return (
+		<React.Fragment>
+			<StyledToolbar>
+				<Title>Organizations</Title>
+				<Search>
+					<IconButton
+						aria-label="search"
+						sx={{
+							color: "#212121",
```

Comment:
Color codes should come css variables, or any other means that allows for it to be dynamically changed by the theme